numba/cuda/compiler.py

```diff
@@ -104,7 +104,10 @@ def compile_cuda(pyfunc, return_type, args, debug=False, inline=False,
     flags.set('no_compile')
     flags.set('no_cpython_wrapper')
     flags.set('no_cfunc_wrapper')
-    if debug:
+    if debug or lineinfo:
+        # Note both debug and lineinfo turn on debug information in the
+        # compiled code, but we keep them separate arguments in case we
+        # later want to overload some other behavior on the debug flag.
```

Comment:
In particular, `-opt=3` is not supported with `-g`, so it's a good thing to keep them separate: https://docs.nvidia.com/cuda/libnvvm-api/group__compilation.html#group__compilation_1g76ac1e23f5d0e2240e78be0e63450346

Reply:
I've added a comment to this effect.